docs/source/entities-advanced.mdx

```diff
@@ -57,6 +57,32 @@ type Organization {
 }
 ```
 
+### Differing `@key`s across subgraphs
+
+An entity often has the same @key fields across subgraphs, but this isn't required. For example, you can define a `Product` entity shared between products and inventory subgraphs, one with `id` as the `@key` field and one with `sku` as the `@key` field:
+
+<CodeColumns>
+
+```graphql title="Products subgraph"
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+```graphql title="Inventory subgraph"
+type Product @key(fields: "sku") {
+  sku: ID!
+  inStock: Boolean!
+}
+```
+
+</CodeColumns>
+
+As long as `id` is a unique identifier within the products subgraph and `sku` is a unique identifier for the same product in the inventory subgraph, the subgraphs can each resolve the `Product` type.
+In other words, **each key declaration is specific to the subgraph it belongs to**, and you don't need to match `@key` fields between graphs.
```

Comment:
To clarify this a bit, the `key` acts as a sort of sign to the query planner that says "hey you! i need this data to be able to identify this thing!". 

Just like talking to them, you'd also need to have the thing they're asking for in order for it to work. In the above example, the query planner wouldn't work. It'd choke because entering from either products or inventory can't ever lead to the other subgraph. 

This is the crux of _why_ we've historically made it so that the keys must match. 

A concrete example of this working would be: 

sub a:
```gql
type Query {
  pi: Product
}

type Product @key(fields: "sku") {
  sku: ID!
  upc: String!
  inStock: Boolean!
}
```

sub b: 
```gql
type Product @key(fields: "upc") {
  upc: String! @external
  name: String!
  price: Int
}
```

But as soon as you add a query to B, it needs to reference _something_ from sub a. It's a really weird edge-case and isn't going to come up tremendously often. 

@pcmanus may be able to explain this or at least confirm I'm not totally wrong here.

Reply:
I guess the most common use case for different `@key`s is that in one subgraph you could have multiple `@key`s and in another subgraph just one of them. 

While technically subgraphs don't have to share the same `@key`, practically 
- if they share the same fields (e.g. `upc` in the example above, which doesn't need the `@external`) - why wouldn't you define it as a `@key` in all subgraphs that have that field (which gives more flexibility to query planner)?
- if they don't share any of the fields - you won't be able to cross subgraph boundaries, i.e. you won't be able to merge entities as there is no way to merge entities using information from just one subgraph. **Queries against single subgraph will work though**

Reply:
This is very helpful, thank you @lleadbet and @dariuszkuc ! Just updated the text based on this info, would you take another look whenever you get a chance?